AppCenterReactNativeShared/Products/local.podspec

```diff
@@ -0,0 +1,18 @@
+# This podspec is intended to be used as local pod reference during development,
```

Comment:
Why do we need a local pod repo? Just curiosity.

Reply:
We need the `local.podspec` with local reference for local development while keeping `AppCenterReactNativeShared.podspec` with official reference for Prod usage.

Prior to RN 0.60, we referencing the file path in the `Podfile` of TestApp like this (https://github.com/microsoft/appcenter-sdk-react-native/blob/2.1.0/TestApp/ios/Podfile#L15):

```ruby
pod 'AppCenterReactNativeShared', :path => '../../AppCenterReactNativeShared/Products/'
```

But with RN 0.60 autolinking, it is not possible to do so. We figured out the easiest way is to add a separate podspec for local development,

```ruby
pod 'AppCenterReactNativeShared', :podspec => '../../AppCenterReactNativeShared/Products/local.podspec'
```

To sum up, in local development, when you run `.\update-npm-packages.sh` from `TestApp` directory, the script calls `pod install` (https://github.com/microsoft/appcenter-sdk-react-native/pull/634/files#diff-8d44ab4d824b97b6ba9ca3cfacc4760eR34) which internally build and install from `local.podspec`.